### DISCUSSION TITLE
Checking when a HitDef connects before they flinch

### DISCUSSION THREAD

**Discussion by Kokorz** [ikemen-engine/Ikemen-GO#1031] [Feature Request]
### Is your feature request related to a problem? Please describe.

Saw a discussion on discord about unhardcoding certain stuff [(here)](https://discord.com/channels/233345562261323776/233363722934943744/1083924284793172119), and I thought hitdef sound/explod stuff was one of the more annoying examples of stuff being so limiting when they shouldnt. I have discussed two solutions in #987, but I think I'd really like to push on the MoveHitTrue idea, along with its equivalents.

### Describe the solution you'd like

Make a trigger that checks if a HitDef has made contact with the opponent immediately, as in the frame before they flinch. Ideally, there should be variants for if it was a hit, a blocked hit, and so on.

An example could be:

`HitDefConnect(hit)` checks if the character has hit the opponent and has not flinched. `HitDefConnect(guarded)` checks for the same but if the attack was guarded. So on, so forth for movereversed, hitoverriden, movecountered, etc.

### Additional context

This trigger should only persist for one frame, and then disable itself as soon as the opponent has flinched. 

Normal movehit only works after the hit applies or the opponent flinches, making it impossible to properly customize hitsparks to a better degree such as angle rotation or alignment (postype).

 This is ultimately the best method if you don't want to touch HitDef as a State Controller.  

### Search terms

trigger movehit moveguarded hitdef

**Comment by potsmugen** [comment]
I think implementing it the exact way you describe is not possible, since the moment before the chars flinch happens _after_ the char code was already executed. Hit detection is checked at the end of the current frame, in other words.

That said, triggers that only return true for one frame are currently implemented as MoveHitVar(Contact) for the attacker and GetHitVar(Contact) for the victim.

**Reply by Kokorz** [reply]
I'm not going to lie... I forgot I opened this discussion. I tried both of those triggers you mentioned, and they work well enough for my purposes, So, I can definitely close this lmao
